### DISCUSSION TITLE
Smart Query to get all entries not referenced in another table based on filtered collection

### DISCUSSION THREAD

**Discussion by Darent98** [mikro-orm/mikro-orm#4253] [General]
**Is your feature request related to a problem? Please describe.**
I could not find a way to write a Smart Query that does the following: Find all entries of an entity A that references an entity B, where the result should only contain those entries of A, that are not referenced in B, with the restriction to only consider entries of B with a specific pathId.
In the example below you can see how I tried to accomplish it, without success.
With *populateWhere* I tried to populate usedB with only entries in the specified pathId and then get the entries of A where no Bs are referenced, but it did not work.

**Describe the solution you'd like**
It would be nice to have the option to write a Smart Query that allows to filter out all entries of A that are already referenced in B. If there is already a way to accomplish this, please let me know.

**Describe alternatives you've considered**
I got it working with a subquery and the QueryBuilder, but not with a Smart Query

**Additional context**
Example:
```@Entity({ tableName: 'A' })
export class A {
  @PrimaryKey()
    id!: number

  @Property()
    deprecated!: boolean

  @OneToMany(() => B, _ => _.A)
    usedB = new Collection<B>(this)
}

@Entity({ tableName: 'B' })
export class B {
  @PrimaryKey()
    id!: number

  @Property()
    pathId!: number

  @ManyToOne({ entity: () => A })
    A!: IdentifiedReference<A>
}

async function getNotAddedAs (pathId: number): Promise<A[]> {
  return await orm.em.find(A, { deprecated: false, usedB: null }, { populate: ['usedB'], populateWhere: { usedB: { pathId } } })
}
```

**Comment by B4nan** [comment]
This sounds like a duplicate of #2916, if not, can you share what exactly do you expect from this - how did you do it with the QB?
